File: qqbot-module-plugin-support/src/main/java/com/mieai/qqbot/admin/plugins/PluginInventoryResponse.java
package com.mieai.qqbot.admin.plugins;

import java.time.Instant;
import java.util.List;
import java.util.Objects;

public record PluginInventoryResponse(
        List<PluginArtifactResponse> items,
        String directory,
        boolean directoryExists,
        boolean runtimeAvailable,
        String scanError,
        Instant scannedAt) {

    public PluginInventoryResponse {
        items = List.copyOf(Objects.requireNonNull(items, "items must not be null"));
        Objects.requireNonNull(directory, "directory must not be null");
        Objects.requireNonNull(scannedAt, "scannedAt must not be null");
    }
}

File: qqbot-module-plugin-support/src/main/kotlin/com/mieai/qqbot/admin/plugins/PluginInventoryResponse.kt
package com.mieai.qqbot.admin.plugins

import java.time.Instant

class PluginInventoryResponse(
    items: List<PluginArtifactResponse>,
    val directory: String,
    val directoryExists: Boolean,
    val runtimeAvailable: Boolean,
    val scanError: String?,
    val scannedAt: Instant,
) {
    val items: List<PluginArtifactResponse> = items.toList()
}
